### DISCUSSION TITLE
Add new image filters/effects from hakuimg

### DISCUSSION THREAD

**Discussion by CuteCat1111** [WASasquatch/was-node-suite-comfyui#98] [Ideas]
Auto1111 has a cool extension for working with images: https://github.com/KohakuBlueleaf/a1111-sd-webui-haku-img/
How about taking the code from there and implementing image filters in comfyUI?

Filters: https://github.com/KohakuBlueleaf/a1111-sd-webui-haku-img/tree/main/hakuimg

**Comment by WASasquatch** [comment]
All of that is in WAS-NS except tone curves, lens, and whatever neon is. Don't really like ripping code either. I try to research commonly used stuff and port the idea to python

**Reply by CuteCat1111** [reply]
OK, thanks for the answer

Some filters for working with color are quite interesting, you can see if you want: https://github.com/KohakuBlueleaf/a1111-sd-webui-haku-img/tree/main/hakuimg/color

**Reply by WASasquatch** [reply]
It's all pretty basic stuff you can do with WAS-NS already like HDR effect which is just adjustments and blending modes. Curves stuff I havent added cause there is no pleasing way to know what you are doing without a front end extension

**Reply by CuteCat1111** [reply]
Ok, thx
